src/ServiceManager.php

```diff
@@ -976,13 +939,31 @@ private function assertCallableDelegatorFactory($delegatorFactory): void
                 . ' which does not exist; please provide a valid function name or class name resolving'
                 . ' to an implementation of %s',
                 $delegatorFactory,
-                Factory\DelegatorFactoryInterface::class
+                DelegatorFactoryInterface::class
             ));
         }
         throw new ServiceNotCreatedException(sprintf(
             'A non-callable delegator, "%s", was provided; expected a callable or instance of "%s"',
             is_object($delegatorFactory) ? $delegatorFactory::class : gettype($delegatorFactory),
-            Factory\DelegatorFactoryInterface::class
+            DelegatorFactoryInterface::class
         ));
     }
+
+    /**
+     * @return DelegatorFactoryInterface|DelegatorCallableType
+     */
+    private function resolveDelegatorFactory(mixed $delegatorFactory): callable|DelegatorFactoryInterface
```

Comment:
Assuming well-formed configuration, can we perhaps be stricter on the input type here?

Reply:
Done that in the recent commits.